### PR TITLE
Port `nll_loss_backward` to structured

### PR DESCRIPTION
**Pull request by SplitInfinity**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #62144

Differential Revision: [D29945279](https://our.internmc.facebook.com/intern/diff/D29945279)

### CODE REVIEW

**Comment by SplitInfinity** [comment]
@splitinfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29945279).

File: aten/src/ATen/native/LossNLL.cpp [hunk 1]
```diff
@@ -51,6 +53,56 @@ TORCH_META_FUNC(nll_loss_forward)
...
+    const auto batch_size = self.size(0);
+    check_dim_size(grad_output, 1, 0, batch_size);
+  } else {
+    TORCH_CHECK(
+        grad_output.dim() <= 1 && grad_output.numel() == 1,
+        "Expected a single element grad_output tensor, but got: ",
+        grad_output.sizes());
+  }
+
+  TensorOptions options(self.options());
```
Comment (ezyang) [inline aten/src/ATen/native/LossNLL.cpp:103]: TensorOptions is entirely functional, so you can do this fluently (`self.options().memory_format(...)`)

**Review by ezyang — APPROVED** [review]

**Comment by SplitInfinity** [comment]
@splitinfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29945279).

File: aten/src/ATen/native/LossNLL.cpp [hunk 2]
```diff
@@ -52,6 +54,55 @@ TORCH_META_FUNC(nll_loss_forward)
...
+      ")")
+  TORCH_CHECK(
+      total_weight.numel() == 1,
+      "expected total_weight to be a  single element tensor, got: ",
+      total_weight.sizes(),
+      " (",
+      total_weight.numel(),
+      " elements)");
+
+  auto weight = weight_opt.getTensorRef();
```
Comment (ezyang) [inline aten/src/ATen/native/LossNLL.cpp:86]: `const auto& weight`

**Review by ezyang — APPROVED** [review]
excellent!

**Comment by SplitInfinity** [comment]
@splitinfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29945279).

**Pull request closed without merging**